newIDE/app/src/Leaderboard/LeaderboardProvider.js

```diff
@@ -186,10 +188,21 @@ const LeaderboardProvider = ({ gameId, children }: Props) => {
 
   const listLeaderboards = React.useCallback(
     async () => {
-      dispatch({ type: 'SET_LEADERBOARDS', payload: null });
-      const fetchedLeaderboards = await listGameLeaderboards(gameId);
-      fetchedLeaderboards.sort((a, b) => a.name.localeCompare(b.name));
-      dispatch({ type: 'SET_LEADERBOARDS', payload: fetchedLeaderboards });
+      if (!isListingLeaderboards.current) {
+        isListingLeaderboards.current = true;
+        const fetchLeaderboardAndDispatch = async () => {
+          dispatch({ type: 'SET_LEADERBOARDS', payload: null });
+          const fetchedLeaderboards = await listGameLeaderboards(gameId);
+          fetchedLeaderboards.sort((a, b) => a.name.localeCompare(b.name));
+          dispatch({
+            type: 'SET_LEADERBOARDS',
+            payload: fetchedLeaderboards,
+          });
+        };
+        fetchLeaderboardAndDispatch().then(
+          () => (isListingLeaderboards.current = false)
```

Comment:
Add a try/finally to be sure that you set `isListingLeaderboards.current` back to false, even if something was thrown